Gem/Code/Source/Weapons/BaseWeapon.cpp

```diff
@@ -20,19 +24,50 @@ namespace MultiplayerSample
         , m_weaponParams(constructParams.m_weaponParams)
         , m_weaponListener(constructParams.m_weaponListener)
     {
-/*
-        @TODO: Need a replacement fx and material fx system
-        ArchetypeAssetLibrary* assetLibsInstance = ArchetypeAssetLibrary::GetInstance();
-        NV_ASSERT(assetLibsInstance != nullptr, "Asset library is NULL");
+#if AZ_TRAIT_CLIENT
+        if (PopcornFX::PopcornFXRequests* popcornFx = PopcornFX::PopcornFXRequestBus::FindFirstHandler())
```

Comment:
Nit: This seems strange that this this is in the BaseWeapon class that its tied to specific Vfx and Audio. To me "BaseWeapon" indicates bare bones framework which through a suitable pattern (ie composition etc) gets the ability to be extended to AVWeapon or similar.

I

Reply:
We need some kind of effects system.. I could take the time to write an interface and then plug in pkfx and wwise to the backend which would make all of this even more slick, but given time constraints I decided to opt for this faster solution instead. Without something like this PR, we'll have to go back to using the pkfx component and working with the constraint of one effect for one entity and communicating all through component busses with specialized components/code like the WeaponEffectComponent used to do.